web/src/main/angular/src/app/core/components/response-avg-max-chart/response-avg-max-chart-container.component.ts

```diff
@@ -0,0 +1,333 @@
+                        catchError(() => of(null)),
+                        filter((res: any) => !!res),
+                        map((data: any) => this.isAllAgent() ? data['responseStatistics'] : data['agentResponseStatistics'][this.selectedAgent])
+                    );
+                }),
+            )
+        ).pipe(
+            map((data: IResponseStatistics) => this.cleanIntermediateChartData(data)),
+            map((data) => this.makeChartData(data)),
+            switchMap((data: PrimitiveArray[]) => {
+                if (this.shouldUpdateYMax()) {
+                    const maxTickValue = getMaxTickValue(data, 1);
+                    const yMax = maxTickValue === 0 ? this.defaultYMax : maxTickValue;
+
+                    this.storeHelperService.dispatch(new Actions.UpdateResponseAvgMaxChartYMax(yMax));
+
+                    return of([data, yMax]);
+                } else {
+                    return of(data).pipe(
+                        withLatestFrom(this.storeHelperService.getResponseAvgMaxChartYMax(this.unsubscribe))
+                    );
+                }
+            }),
+        ).subscribe(([chartData, yMax]: [PrimitiveArray[], number]) => {
+            this.chartConfig = {
+                dataConfig: this.makeDataOption(chartData),
+                elseConfig: this.makeElseOption(yMax),
+            };
+
+            this.cd.markForCheck();
+        });
+    }
+
+    private shouldUpdateYMax(): boolean {
+        return this.isAllAgent() && this.isOriginalNode;
+    }
+
+    private isAllAgent(): boolean {
+        return this.selectedAgent === '';
+    }
+
+    private getTargetInfo(): any {
+        return this.selectedTarget.isNode
+            ? this.serverMapData.getNodeData(this.selectedTarget.node[0])
+            : this.serverMapData.getLinkData(this.selectedTarget.link[0]);
+    }
+
+    private cleanIntermediateChartData(data: IResponseStatistics): IResponseStatistics {
+        return data ? filterObj((key: string) => key === 'Max' || key === 'Avg', data) : data;
+    }
+
+    private makeChartData(data: IResponseStatistics): PrimitiveArray[] {
+        return data
+            ? [['x', ...Object.keys(data)], ['rs', ...Object.values(data)]]
+            : [];
+    }
+
+    private makeDataOption(columns: PrimitiveArray[]): Data {
+        return {
+            x: 'x',
+            columns,
+            empty: {
+                label: {
+                    text: this.dataEmptyText
+                }
+            },
+            type: 'bar',
+            color: (_, {index}: DataItem): string => this.chartColors[index],
+            labels: {
+                colors: '#333',
+                format: {
+                    rs: (v: number) => this.convertWithUnit(v)
+                }
+            }
+        };
+    }
+
+    private makeElseOption(yMax: number): {[key: string]: any} {
+        return {
+            padding: {
+                top: 20
+            },
```

Comment:
You may want to add some padding on the right side because the right-most x-tick is cut
```suggestion
                top: 20,
                right: 20  // (or 25)
            },
```